feat: 데일리노트 조회 API

Chedda98 opened the pull request:
<!-- 
- PR 제목: [feat/fix/refactor...] 작업 내용 한 줄 요약 (브랜치 이름)
- commit message가 적절한지 확인해주세요.
- 적절한 branch로 요청했는지 확인해주세요.
- Assignees, Label을 붙여주세요.
- 주의 사항과 관련해 꼭 확인해야 할 사람이 있다면 Reviewer로 등록해주세요.
- PR이 승인된 경우 해당 브랜치는 삭제 부탁드립니다. 
-->


## 📝 PR 요약
[작업중] 데일리노트 조회 API입니다. 
<img width="606" alt="스크린샷 2023-01-23 오전 1 52 55" src="https://user-images.githubusercontent.com/68318945/213928173-f03c70d1-4ec1-43f5-843f-f7147250d9f7.png">

<br/>

## 📌 변경 사항 및 주의 사항

<!--
변경사항 및 주의 사항이 있다면 적어주세요.
주의 사항과 관련해 꼭 확인해야 할 사람이 있다면 리뷰어로 등록해주세요. (다른 사람이 작성한 코드 수정 등)
코드 리뷰 시 더 꼼꼼하게 확인 받고 싶은 부분이 있다면 적어주세요.
-->
- **`planDate` type이 변경**되었는데 이 부분 코드 내에 멘션 걸어두겠습니다!!

이전에는 날짜를 쿼리로 받아 해당 날짜의 데일리노트를 보여주는 형태인 것으로 확인되는데,
- 기존과 동일하게 쿼리로 날짜를 받아 해당 날짜의 데일리노트 결과값 출력 / 없다면 널값 출력 -> 이렇게 가기로 했습니다!!
- _`userId`를 통해 해당 유저의 데일리노트 모두 조회 (날짜순)_

<br/>

## 🔗 Linked Issue
close #10

m1njae committed:
feat: 소셜 로그인 및 회원 가입 API

토큰을 받지 않아 전반적인 로직만 작성해놓은 상태입니다. 토큰을 받아 API 테스트하면서 발생하는 오류들을 수정하겠습니다.

m1njae committed:
feat: 토큰 재발급

m1njae committed:
feat: 계획블록 이름 및 컬러칩 수정

DB 수정의 이유로 작업 중이었던 부분 잠시 홀딩하고 merge 후 재작업하도록 하겠습니다.

m1njae committed:
chore: entities 수정 적용

m1njae committed:
fix: updatePlans 로직 수정

m1njae committed:
fix: updatePlans 로직 수정

m1njae committed:
chore: 계획블록 수정 responsMessage 변경

m1njae committed:
chore: 계획이름, 컬러칩 이외에 다른 값이 들어올 경우 예외처리

m1njae committed:
chore: class명 변경

m1njae committed:
chore: entities 수정 적용

m1njae committed:
add: responseMessage 추가

m1njae committed:
feat: 계획 블록 삭제 구현

Chedda98 committed:
feat: 데일리노트 조회 구조 셋팅

m1njae committed:
chore: 코드리뷰 피드백 반영

3개의 if 조건문을 사용했던 이전 코드와는 달리 조건문 2개로 로직을 수정하였습니다.

Chedda98 committed:
feat: 데일리노트 조회 기능 구현

Chedda98 committed:
add: 데일리노트 조회 response message 추가 및 적용

m1njae committed:
feat: 계획 블록 수정하기 시, 우회 취소 구현

m1njae committed:
add: dayjs 설치

m1njae committed:
chore: responseMessage 추가

m1njae committed:
chore: planDate 타입 변경

m1njae committed:
feat: 계획블록 생성

m1njae committed:
fix: type이 routine일 때 로직 수정

Chedda98 committed:
fix: 데일리노트 createdAt 컬럼 속성

m1njae committed:
chore: userId type 변경

구글 회원번호가 integer 범위를 넘어가 급한대로  type을 string으로 변경해놓았습니다.

m1njae committed:
fix: 소셜 로그인 및 회원가입, 토큰 재발급 로직 수정

m1njae committed:
chore: userId type string에서 bigint로 변경

Chedda98 committed:
Fix: planDate type 변경 및 데일리노트 조회 API 완료

m1njae committed:
chore: transaction 삭제 및 save 메소드 insert로 변경

Chedda98 commented on src/entities/DailyNote.ts:
```diff
@@ -12,8 +12,8 @@ export class DailyNote {
-  @CreateDateColumn({ type: 'timestamp', name: 'created_at', nullable: false })
-  createdAt?: Date;
+  @Column({ name: 'plan_date', nullable: false })
+  planDate?: string;
```
@dannaward cc @m1njae 
민재와는 구두로 논의를 했는데 다나와도 논의가 필요해서 멘션 남겨둡니다!!

사실 드래그미 서비스의 경우, 계획하는 행위에 있어 계획을 만드는 시점을 저장하는 것이 중요한게 아니라 어떤 날의 계획을 세웠는지가 중요한 것 같습니다! (예를 들어, 이번주 일요일에 등산가고, 네일하고, 드래그미 작업하는 계획을 당일에 그 계획을 행할 때 만드는 것이 아니라 전날인 토요일 또는 그보다 더 이전에 계획을 생성하는 것)

그래서 `planDate`의 정의를 조금 바꿔서 "_계획을 생성한 date_"가 아닌 "_계획을 세운 그 날_"에 초점을 맞추고자 type을 **Date가 아닌 string**으로 바꾸면 어떨지 논의드려보고 싶습니다 !!

m1njae committed:
chore: insert 메소드 save으로 변경

insert로 변경 시, 반환타입이 InsertResult가 되어 로직에 문제가 생깁니다.
단순히 테이블에 값을 추가하는 경우에는 insert 메소드를 사용하되,
생성한 값을 가지고 로직을 이어나가야 하는 경우에는 save 메소드를 사용하도록 하겠습니다.

m1njae committed:
chore: 코드리뷰 피드백 반영

m1njae committed:
feat: 계획블록 순서 이동 및 변경

m1njae committed:
chore: API 테스트 후, 코드 수정

m1njae committed:
chore: 일간계획 <-> 루틴로드 이동 시, 최종 배열 변경 로직 수정

m1njae commented on src/controllers/DailyNoteController.ts:
```diff
@@ -36,13 +30,11 @@ export class DailyNoteController {
     @Res() res: Response,
     @Param('userId') userId: string,
-    @QueryParams() query: GetTypeAndDateQuery,
+    @QueryParams() planDate: string,
```
쿼리 파라미터가 하나만 있을 경우에는 `@QueryParam('planDate') planDate: string` 으로 작성하면 조금 더 코드가 깔끔해질거 같다는 생각이 들어요!

m1njae commented on src/controllers/DailyNoteController.ts:
```diff
@@ -36,13 +30,11 @@ export class DailyNoteController {
-        query.createdAt,
-      );
+      const data = await this.dailyNoteService.getDailyNote(+userId, planDate);
+      console.log(data);
```
열심히 작업한 흔적이 담겨있는 `console.log()` 이젠 보내줄 때가 된 것 같습니다ㅎㅎ

m1njae reviewed (approved):
체다 바쁜데 고생 많이 했어요~ 감사합니다(__) 멘션 걸어둔 거 다나도 OK하면 머지하도록 해요😊👍

m1njae committed:
chore: refreshtoken 만료주기 30일로 변경

m1njae committed:
Merge pull request #25 from Team-DragMe/feature/15

feat: 소셜 로그인 및 회원 가입, 토큰 재발급

m1njae committed:
Merge branch 'main' of https://github.com/Team-DragMe/Drop-the-BE into feature/16

m1njae committed:
add: express validator 적용

m1njae committed:
chore: auth 로직 수정

얻어낸 userId를 res.locals.JwtPayload에 담고 미들웨어로 넘깁니다.

m1njae committed:
chore: snsId type 변경

구글 회원번호가 bigInt타입의 범위도 넘어가서 string으로 변경하였습니다.

m1njae committed:
chore: planDate nullable true로 변경

우회할 계획 시, 날짜와 상관없이 유지가 되어야하므로 nullable을 true로 변경하였습니다.

m1njae committed:
chore: isCompleted 컬럼 추가

계획 블록 완료/미완료를 판단하기 위해서 컬럼을 추가하였습니다.

m1njae committed:
fix: 계획블록 조회 및 수정 로직 수정

auth와 errorValidator를 추가하였습니다.
계획블록 수정하기에는 계획블록 완료하는 로직도 추가하였습니다.

m1njae committed:
Merge pull request #20 from Team-DragMe/feature/16

feat: 계획블록 수정 API

m1njae committed:
chore: merge 이후 코드 정리

m1njae committed:
chore: errorValidator 주석 처리

body 값을 유효성 검사하는 과정에서 계속해서 errorValidator가 동작하지 않습니다.
저의 개인적인 생각은 class-transform 하는 과정 때문에 동작하지 않는 것 같은데,
당장은 시간이 부족하니 에러 핸들링을 if 조건문으로 구현하였습니다.

m1njae committed:
chore: planController 코드 정리

m1njae committed:
chore: planService 로직 수정

루틴 로드의 경우 우회할 계획과 마찬가지로 날이 바뀌어도 유지가 되어야 하므로
시간 데이터를 저장하지 않습니다.

m1njae committed:
Merge pull request #24 from Team-DragMe/feature/22

feat: 계획블록 생성

m1njae committed:
chore: merge 이후 코드 정리

m1njae committed:
add: deletePlanValidation 추가

m1njae committed:
chore: planController 코드 정리

m1njae committed:
chore: planService 로직 수정

daily 시에는 planDate 값이 필요로 하지만, 시간 데이터를 저장하지 않고 있는 루틴로드와 우회할 계획의 경우,
시간 데이터가 필요하지 않습니다. 따라서 분기처리하여 로직을 수정하였습니다.

m1njae committed:
Merge pull request #21 from Team-DragMe/feature/17

feat: 계획 블록 삭제

m1njae committed:
chore: merge 이후 코드 정리

m1njae committed:
add: deletePlanValidtion 추가

m1njae committed:
chore: planController 코드 정리

m1njae committed:
chore: planService 로직 수정

시간 데이터 존재 유무에 따라 로직을 분기하였습니다

m1njae committed:
Merge pull request #29 from Team-DragMe/feature/28

feat: 계획블록 순서 이동 및 변경

m1njae committed:
chore: merge 이후 코드 정리

m1njae committed:
add: dailyNoteValidation 추가

m1njae committed:
chore: planDate 타입 변경

m1njae committed:
chore: console.log 삭제

m1njae committed:
chore: typeorm synchronize 설정

m1njae committed:
chore: dailyNoteController 코드 정리
